src/node/types.ts

```diff
@@ -39,3 +43,5 @@ export interface PostCSSConfigResult {
   options?: Postcss.ProcessOptions
   plugins: Postcss.Plugin[]
 }
+
+export type GetViteConfigFn = (...args: ['build', BuildOptions] | ['serve', ServeOptions]) => InlineConfig
```

Comment:
Do we explicitly need to type that function here besides generic, reusable Types? I think inlining it to the function would be sufficient, since it's not re-used anywhere else.